src/axom/quest/Shaper.cpp

```diff
@@ -30,13 +27,133 @@ constexpr double Shaper::MINIMUM_PERCENT_ERROR;
 constexpr double Shaper::MAXIMUM_PERCENT_ERROR;
 constexpr double Shaper::DEFAULT_VERTEX_WELD_THRESHOLD;
 
-Shaper::Shaper(const klee::ShapeSet& shapeSet, sidre::MFEMSidreDataCollection* dc)
-  : m_shapeSet(shapeSet)
+#if defined(AXOM_USE_MFEM)
+Shaper::Shaper(RuntimePolicy execPolicy,
+               int allocatorId,
+               const klee::ShapeSet& shapeSet,
+               sidre::MFEMSidreDataCollection* dc)
+  : m_execPolicy(execPolicy)
+  , m_allocatorId(allocatorId != axom::INVALID_ALLOCATOR_ID
+                    ? allocatorId
+                    : axom::policyToDefaultAllocatorID(execPolicy))
+  , m_shapeSet(shapeSet)
   , m_dc(dc)
+  #if defined(AXOM_USE_CONDUIT)
+  , m_bpGrp(nullptr)
+  , m_bpTopo()
+  , m_bpNodeExt(nullptr)
+  , m_bpNodeInt()
+  #endif
 {
-#if defined(AXOM_USE_MPI) && defined(MFEM_USE_MPI)
+  #if defined(AXOM_USE_MPI) && defined(MFEM_USE_MPI)
   m_comm = m_dc->GetComm();
+  #endif
+  m_cellCount = m_dc->GetMesh()->GetNE();
+
+  setFilePath(shapeSet.getPath());
+}
+#endif
+
+Shaper::Shaper(RuntimePolicy execPolicy,
+               int allocatorId,
+               const klee::ShapeSet& shapeSet,
+               sidre::Group* bpGrp,
+               const std::string& topo)
+  : m_execPolicy(execPolicy)
+  , m_allocatorId(allocatorId != axom::INVALID_ALLOCATOR_ID
+                    ? allocatorId
+                    : axom::policyToDefaultAllocatorID(execPolicy))
+  , m_shapeSet(shapeSet)
+#if defined(AXOM_USE_CONDUIT)
+  , m_bpGrp(bpGrp)
+  , m_bpTopo(topo.empty() ? bpGrp->getGroup("topologies")->getGroupName(0) : topo)
+  , m_bpNodeExt(nullptr)
+  , m_bpNodeInt()
+#endif
+  , m_comm(MPI_COMM_WORLD)
+{
+  SLIC_ASSERT(m_bpTopo != sidre::InvalidName);
+
+  // This may take too long if there are repeated construction.
+  m_bpGrp->createNativeLayout(m_bpNodeInt);
+
+#if defined(AXOM_DEBUG)
+  std::string whyBad;
+  bool goodMesh = verifyInputMesh(whyBad);
+  SLIC_ASSERT_MSG(goodMesh, whyBad);
+#endif
+
+  m_cellCount = conduit::blueprint::mesh::topology::length(
+    m_bpNodeInt.fetch_existing("topologies").fetch_existing(m_bpTopo));
+
+  setFilePath(shapeSet.getPath());
+}
+
+Shaper::Shaper(RuntimePolicy execPolicy,
+               int allocatorId,
+               const klee::ShapeSet& shapeSet,
+               conduit::Node* bpNode,
+               const std::string& topo)
+  : m_execPolicy(execPolicy)
+  , m_allocatorId(allocatorId != axom::INVALID_ALLOCATOR_ID
+                    ? allocatorId
+                    : axom::policyToDefaultAllocatorID(execPolicy))
+  , m_shapeSet(shapeSet)
+#if defined(AXOM_USE_CONDUIT)
```

Comment:
Does the `#ifdef` need to extend for the whole constructor, with `bpNode`?

Reply:
The `#ifdef` guards in the shaper classes are a bit messy.  @BradWhitlock and I placed them as needed while trying to avoid affecting other parts of the code.  We do need to rethink the dependencies at the class, test code and CMake levels.  This has been added to the follow-up issue https://github.com/LLNL/axom/issues/1445.